numpy/core/tests/test_simd.py

```diff
@@ -632,6 +721,32 @@ def test_arithmetic_reduce_sum(self):
         vsum = self.sum(vdata)
         assert vsum == data_sum
 
+    def test_mask_conditional(self):
+        """
+        Conditional addition and subtraction for all supported data types.
+        Samples:
+            npyv_s32 npyv_ifadd_s32(npyv_b32 mask, npyv_s32 a, npyv_s32 b, npyv_s32 c) ->
+                mask ? a + b : c
+            npyv_f64 npyv_ifsub_f64(npyv_b64 mask, npyv_f64 a, npyv_f64 b, npyv_f64 c) ->
+                mask ? a - b : c
```

Comment:
@mattip, Is that okay?

Reply:
I was thinking along the lines of just

Test npyv_ifadd_##SFX and npyvv_ifsub_##SFX

That way it is easy to grep the tests and header files for coverage without needing too much added prose in the test

Reply:
something like that:
```Python
    def test_mask_conditional(self):
        """
        Test the following intrinsics:
        
            - npyv_ifadd_u8,  npyv_ifadd_s8,  npyv_ifadd_s16, npyv_ifadd_u16, npyv_ifadd_u32,
              npyv_ifadd_s32, npyv_ifadd_u64, npyv_ifadd_s64, npyv_ifadd_f32, npyv_ifadd_f64

            - npyv_ifsub_u8,  npyv_ifsub_s8,  npyv_ifsub_s16, npyv_ifsub_u16, npyv_ifsub_u32,
              npyv_ifsub_s32, npyv_ifsub_u64, npyv_ifsub_s64, npyv_ifsub_f32, npyv_ifsub_f64
        """
````

Reply:
No, I meant exactly `npyv_ifadd_##SFX` since that is the grep-able macro name. The name `npyv_ifadd_u8` does not appear in the source code.

Reply:
> The name npyv_ifadd_u8 does not appear in the source code.

Because its an inline function generated by C macro

Reply:
defined in [avx512/maskop.h](https://github.com/numpy/numpy/blob/d7a183ed821e6e576e508ae8ecb82332263ed04d/numpy/core/src/common/simd/avx512/maskop.h) and [emulate_maskop.h](https://github.com/numpy/numpy/blob/d7a183ed821e6e576e508ae8ecb82332263ed04d/numpy/core/src/common/simd/emulate_maskop.h)

Reply:
I think a docstring that mentions the actual C macro `npyv_ifadd_##SFX` would be more helpful in linking the test to the code it is testing.

Reply:
I literally followed your suggestion but I don't think it would for a useful coverage test. I think there's a possibility to generate  [_simd.dispatch.c.src](https://github.com/numpy/numpy/blob/master/numpy/core/src/_simd/_simd.dispatch.c.src) directly from the NPYV headers including the doc str. Is that would be a good idea?

Reply:
Maybe in a future PR. I think I would still like the resulting generated test code to be checked in, so we can read it. Otherwise it will be very painful to track down errors in tests.